Docs/fix lb multicomponent

rzaman64 opened the pull request:
Adds new documentation file fix_lb_multicomponent.rst describing the ternary lattice Boltzmann model implemented in fix_lb_multicomponent.cpp.
This fix extends fix_lb_fluid to support three-component mixtures, following the work by Arumugam Kumar et al. (2024).

Related Issue(s)
N/A – new documentation entry for existing code developed under Schiller Lab.

Author(s)
Rasib Zaman ([rzaman@udel.edu](mailto:rzaman@udel.edu)
), Schiller Lab – University of Delaware

Licensing
I agree that my contribution will be included in LAMMPS and redistributed under GPL v2 / LGPL v2.1.

Backward Compatibility
No changes to code or input behavior — documentation only.

Implementation Notes
Documentation tested with the LAMMPS Sphinx build system and formatted consistently with existing fix_lb_fluid.rst.

Post Submission Checklist

 Documentation is still in progress

 Licensing information is complete

 The doc file builds correctly with the LAMMPS documentation system

rzaman64 committed:
Add initial fix_lb_multicomponent documentation

rzaman64 committed:
Add documentation for fix_lb_multicomponent

rzaman64 committed:
Reformatted to RST, added Boundary conditions.

rzaman64 committed:
Add documentation for fix_lb_multicomponent

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+* lb/multicomponent = style name of this fix command
+* nevery = update the lattice-Boltzmann fluid every this many timesteps (should normally be 1)
+* viscosity = the fluid viscosity (units of mass/(time\*length))
+* density = the initial fluid density
```
```suggestion
* density = the fluid density
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+* nevery = update the lattice-Boltzmann fluid every this many timesteps (should normally be 1)
+* viscosity = the fluid viscosity (units of mass/(time\*length))
+* density = the initial fluid density
+* D3Q19 = velocity lattice model (required)
```
```suggestion
* D3Q19 = the only velocity vector set used in this fix (mandatory to mention)
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+           N = output fluid fields every N timesteps
+           filename = base name for output files (.xdmf and .raw extensions added automatically)
+           timeI = 1 (use simulation time to index xdmf file), 0 (use output frame number)
+       *seed* value = random number generator seed (positive integer)
```
```suggestion
       *seed* value = random number generator seed (positive integer) - useful to initialize certain models like ternary mixture, ternary film, and ternary droplet 
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+  .. parsed-literal::
+
+       *tau_r* value = relaxation time for momentum transport (related to viscosity)
```
```suggestion
       *tau_r* value = relaxation time constant for the population distribution *f* - related to viscosity
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+  .. parsed-literal::
+
+       *tau_r* value = relaxation time for momentum transport (related to viscosity)
+       *tau_p* value = relaxation time for order parameter φ transport
```
```suggestion
       *tau_p* value = relaxation time constant for the population distribution *g* - related to the mobility coefficient of the order parameter $\phi$
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+       *tau_r* value = relaxation time for momentum transport (related to viscosity)
+       *tau_p* value = relaxation time for order parameter φ transport
+       *tau_s* value = relaxation time for order parameter ψ transport
```
```suggestion
       *tau_s* value = relaxation time constant for the population distribution *k* - related to the mobility coefficient of the order parameter $\psi$
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *tau_r* value = relaxation time for momentum transport (related to viscosity)
+       *tau_p* value = relaxation time for order parameter φ transport
+       *tau_s* value = relaxation time for order parameter ψ transport
+       *kappa1* value = surface tension parameter between components C1 and C2
```
```suggestion
       *kappa1* value = energy gradient parameter for fluid 1 - used in surface tension calculation between fluid 1 and other fluids
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *tau_p* value = relaxation time for order parameter φ transport
+       *tau_s* value = relaxation time for order parameter ψ transport
+       *kappa1* value = surface tension parameter between components C1 and C2
+       *kappa2* value = surface tension parameter between components C1 and C3
```
```suggestion
       *kappa2* value = energy gradient parameter for fluid 2 - used in surface tension calculation between fluid 2 and other fluids
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *tau_s* value = relaxation time for order parameter ψ transport
+       *kappa1* value = surface tension parameter between components C1 and C2
+       *kappa2* value = surface tension parameter between components C1 and C3
+       *kappa3* value = surface tension parameter for component C3
```
```suggestion
       *kappa3* value = energy gradient parameter for fluid 3 - used in surface tension calculation between fluid 3 and other fluids
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *kappa1* value = surface tension parameter between components C1 and C2
+       *kappa2* value = surface tension parameter between components C1 and C3
+       *kappa3* value = surface tension parameter for component C3
+       *alpha* value = gradient energy coefficient
```
```suggestion
       *alpha* value = parameter related to the interface width measurement
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *kappa2* value = surface tension parameter between components C1 and C3
+       *kappa3* value = surface tension parameter for component C3
+       *alpha* value = gradient energy coefficient
+       *gamma_p* value = mobility coefficient for order parameter φ
```
```suggestion
       *gamma_p* value = mobility coefficient for order parameter $\phi$
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *kappa3* value = surface tension parameter for component C3
+       *alpha* value = gradient energy coefficient
+       *gamma_p* value = mobility coefficient for order parameter φ
+       *gamma_s* value = mobility coefficient for order parameter ψ
```
```suggestion
       *gamma_s* value = mobility coefficient for order parameter $\psi$
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *C1* value = bulk concentration of component 1
+       *C2* value = bulk concentration of component 2
+       *C3* value = bulk concentration of component 3 (note: C1 + C2 + C3 = 1)
+       *dumpxdmf* values = N filename timeI
```
```suggestion
       *dumpxdmf* values = N filename
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *dumpxdmf* values = N filename timeI
+           N = output fluid fields every N timesteps
+           filename = base name for output files (.xdmf and .raw extensions added automatically)
+           timeI = 1 (use simulation time to index xdmf file), 0 (use output frame number)
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *seed* value = random number generator seed (positive integer)
+       *init* values = initialization_method [method_parameters]
+           initialization_method = *mixture* or *droplet* or *liquid_lens* or *double_emulsion* or *film* or *mixed_droplet*
+               *mixture* = homogeneous mixture with random fluctuations
```
```suggestion
               *mixture* = well-mixed fluid mixture
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *alpha* value = gradient energy coefficient
+       *gamma_p* value = mobility coefficient for order parameter φ
+       *gamma_s* value = mobility coefficient for order parameter ψ
+       *C1* value = bulk concentration of component 1
```
```suggestion
       *C1* value = bulk concentration of fluid component 1
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *gamma_p* value = mobility coefficient for order parameter φ
+       *gamma_s* value = mobility coefficient for order parameter ψ
+       *C1* value = bulk concentration of component 1
+       *C2* value = bulk concentration of component 2
```
```suggestion
       *C2* value = bulk concentration of fluid component 2
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *gamma_s* value = mobility coefficient for order parameter ψ
+       *C1* value = bulk concentration of component 1
+       *C2* value = bulk concentration of component 2
+       *C3* value = bulk concentration of component 3 (note: C1 + C2 + C3 = 1)
```
```suggestion
       *C3* value = bulk concentration of fluid component 3 (note: C1 + C2 + C3 = 1)
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+       *init* values = initialization_method [method_parameters]
+           initialization_method = *mixture* or *droplet* or *liquid_lens* or *double_emulsion* or *film* or *mixed_droplet*
+               *mixture* = homogeneous mixture with random fluctuations
+               *droplet* radius = binary droplet (C1 and C2) of given radius in C3 solvent
```
```suggestion
               *droplet* radius = droplet in a solvent model - droplet of fluid component 1 of given radius surrounded by fluid component 2 - droplet center is located at the center of the box
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+.. versionadded:: 2024
+
+The **fix lb/multicomponent** command implements a ternary lattice Boltzmann model (LBM) for simulating multicomponent fluid mixtures in three dimensions. This fix is an extension of the single-component :doc:`fix lb/fluid <fix_lb_fluid>`, designed to capture more complex interfacial and phase behaviors such as droplets, emulsions, and films involving three interacting fluid components.
```
```suggestion
The **fix lb/multicomponent** command implements a ternary lattice Boltzmann model (LBM) for simulating three-dimensional multicomponent fluid systems. This fix is an extension of the single-component :doc:`fix lb/fluid <fix_lb_fluid>`, designed to capture complex interfacial and phase behaviors of three immiscible fluid components for various fluid models like ternary mixture, droplets, and films.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The **fix lb/multicomponent** command implements a ternary lattice Boltzmann model (LBM) for simulating multicomponent fluid mixtures in three dimensions. This fix is an extension of the single-component :doc:`fix lb/fluid <fix_lb_fluid>`, designed to capture more complex interfacial and phase behaviors such as droplets, emulsions, and films involving three interacting fluid components.
+
+The algorithm evolves three coupled distribution functions over a D3Q19 lattice and models thermodynamic interactions via chemical potentials and free-energy gradients. The model is suitable for studying phase separation, interface dynamics, droplet stability, and multiphase coexistence.
```
```suggestion
The algorithm evolves three distribution functions over a D3Q19 lattice and models thermodynamic interactions using the Landau-Ginzburg free-energy. This model is suitable for studying multi-component fluid interactions in the aspect of phase separation and interface dynamics.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+**Model Overview**
+
+The ternary fluid composition is specified by three concentration fields C₁, C₂, and C₃, which are related to conserved order parameters:
```
```suggestion
The ternary fluid composition is specified by three concentration fields $C_1$, $C_2$, and $C_3$, which are related to conserved order parameters:
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+   \phi &= C_1 - C_2 \\
+   \psi &= C_3
+
+where ρ is the total density (approximately constant for incompressible flow), and φ and ψ are order parameters that distinguish the three components.
```
```suggestion
where $\rho$ is the total density (approximately constant for incompressible flow), $\phi$ and \psi$ are order parameters that distinguish the three components.
```

rzaman64 committed:
Apply suggestions from code review

Commiting post pull request to improve clarity and technical accuracy of the documentation

Co-authored-by: Gokul Raman <53153074+akgokulraman@users.noreply.github.com>

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
````diff
@@ -0,0 +1,245 @@
+   \psi &= C_3
+   ```
+
+with :math:`C_1, C_2, C_3` denoting component concentrations.
````
```suggestion
with $\tau_\rho, $\tau_\phi$, and $\tau_psi$ denoting the time constants for individual LBM equations, and has relations to macroscopic properties of viscosity and mobility coefficients.
```

akgokulraman replied:
Include equations talking about time constants and how they are related to fluid properties like viscosity and mobility coefficients.

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
````diff
@@ -0,0 +1,245 @@
+.. math::
+
+   ```
+   \rho &= C_1 + C_2 + C_3 = 1 \\
````
change to the LBM time evolution equations for f, g, h and remove rho, phi, psi

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The *init* keyword specifies how the ternary fluid is initialized:
+
+* **mixture**: Creates a homogeneous mixture with bulk concentrations C1, C2, C3 plus small random fluctuations (±0.01) to seed phase separation. This is the default initialization if no *init* keyword is specified.
```
```suggestion
* **mixture**: Creates a fluid mixture with bulk concentrations C1, C2, C3. This is the default initialization if no *init* keyword is specified.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **mixture**: Creates a homogeneous mixture with bulk concentrations C1, C2, C3 plus small random fluctuations (±0.01) to seed phase separation. This is the default initialization if no *init* keyword is specified.
+
+* **droplet** *radius*: Creates a binary droplet composed of equal parts C1 and C2 (C3 = 0) with specified radius (in lattice units) centered in the simulation box, surrounded by pure C3.
```
```suggestion
* **droplet** *radius*: Creates a binary droplet composed of components 1 and 2 with concentration C1 and C2 (C1+C2=1) with specified radius (in lattice units) centered in the simulation box, surrounded by component 3 in bulk.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **alpha**: Gradient energy coefficient.
+
+* **C1**, **C2**, **C3**: Bulk concentrations of the three components (must sum to 1.0).
```
```suggestion
* **C1**, **C2**, **C3**: Bulk concentrations of the three components.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **droplet** *radius*: Creates a binary droplet composed of equal parts C1 and C2 (C3 = 0) with specified radius (in lattice units) centered in the simulation box, surrounded by pure C3.
+
+* **liquid_lens** *radius*: Creates a lens-shaped droplet of pure C3 with specified radius positioned at the interface between horizontal layers of pure C1 (top half) and pure C2 (bottom half).
```
```suggestion
* **liquid_lens** *radius*: Creates a lens-shaped droplet of component 3 with equal upper and lower radius of curvature, positioned at the interface between horizontal layers of component 1 (top half) and component 2 (bottom half).
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **liquid_lens** *radius*: Creates a lens-shaped droplet of pure C3 with specified radius positioned at the interface between horizontal layers of pure C1 (top half) and pure C2 (bottom half).
+
+* **double_emulsion** *radius*: Creates a Janus-type double emulsion with a droplet of specified radius split into two hemispheres: left hemisphere of pure C1 and right hemisphere of pure C2, surrounded by pure C3 solvent.
```
```suggestion
* **double_emulsion** *radius*: Creates a Janus-type double emulsion with a droplet of specified radius split into two hemispheres: left hemisphere of component 1 and right hemisphere of component 2, centered in the simulation box and surrounded by component 3 in bulk.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **double_emulsion** *radius*: Creates a Janus-type double emulsion with a droplet of specified radius split into two hemispheres: left hemisphere of pure C1 and right hemisphere of pure C2, surrounded by pure C3 solvent.
+
+* **film** *thickness* *C1_film* *C2_film*: Creates a horizontal film of specified thickness (as a fraction of the box height) centered in the simulation box with composition C1_film, C2_film, C3_film = 1 - C1_film - C2_film, surrounded by bulk mixture. Small random fluctuations (±0.01) are added to seed phase separation.
```
```suggestion
* **film** *thickness* *C1_film* *C2_film*: Creates a horizontal film of specified thickness (as a fraction of the box height) centered in the simulation box with composition C1_film, C2_film, C3_film (C1_film + C2_film + C3_film = 1), surrounded by a bulk mixture of C1, C2, C3 (C1+C2+C3 = 1).
```

akgokulraman replied:
We can phrase "Small random fluctuations (±0.01) are added to seed phase separation" a bit differently, without being technically incorrect. Need to discuss.

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **film** *thickness* *C1_film* *C2_film*: Creates a horizontal film of specified thickness (as a fraction of the box height) centered in the simulation box with composition C1_film, C2_film, C3_film = 1 - C1_film - C2_film, surrounded by bulk mixture. Small random fluctuations (±0.01) are added to seed phase separation.
+
+* **mixed_droplet** *radius* *C1_drop* *C2_drop*: Creates a ternary droplet with specified radius and composition (C1_drop, C2_drop, C3_drop = 1 - C1_drop - C2_drop) surrounded by pure C3 solvent. Small random fluctuations (±0.01) are added within the droplet.
```
```suggestion
* **mixed_droplet** *radius* *C1_drop* *C2_drop*: Creates a ternary droplet with specified radius and composition (C1_drop, C2_drop, C3_drop = 1 - C1_drop - C2_drop) surrounded by component 3 in bulk.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **mixed_droplet** *radius* *C1_drop* *C2_drop*: Creates a ternary droplet with specified radius and composition (C1_drop, C2_drop, C3_drop = 1 - C1_drop - C2_drop) surrounded by pure C3 solvent. Small random fluctuations (±0.01) are added within the droplet.
+
+.. math::
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+* **mixed_droplet** *radius* *C1_drop* *C2_drop*: Creates a ternary droplet with specified radius and composition (C1_drop, C2_drop, C3_drop = 1 - C1_drop - C2_drop) surrounded by pure C3 solvent. Small random fluctuations (±0.01) are added within the droplet.
+
+.. math::
+
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+.. math::
+
+   \text{subNbx} = \frac{N_x}{P_x} + 2h_x \\
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+.. math::
+
+   \text{subNbx} = \frac{N_x}{P_x} + 2h_x \\
+   \text{subNby} = \frac{N_y}{P_y} + 2h_y \\
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+   \text{subNbx} = \frac{N_x}{P_x} + 2h_x \\
+   \text{subNby} = \frac{N_y}{P_y} + 2h_y \\
+   \text{subNbz} = \frac{N_z}{P_z} + 2h_z
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+   \text{subNby} = \frac{N_y}{P_y} + 2h_y \\
+   \text{subNbz} = \frac{N_z}{P_z} + 2h_z
+
+where hₓ = hᵧ = hᵤ = 2 is the halo extent required for gradient and Laplacian calculations.
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+This fix accepts several internal parameters that control the thermodynamic and transport properties:
+
+* **tau_r**, **tau_p**, **tau_s**: Relaxation times for momentum transport and order parameter transport. These control the viscosity and diffusivity of the fluid components.
```
```suggestion
* **tau_r**, **tau_p**, **tau_s**: Relaxation times for momentum transport and order parameter transport.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* **tau_r**, **tau_p**, **tau_s**: Relaxation times for momentum transport and order parameter transport. These control the viscosity and diffusivity of the fluid components.
+
+* **gamma_p**, **gamma_s**: Mobility coefficients controlling the rate of change of the phase variables φ and ψ.
```
```suggestion
* **gamma_p**, **gamma_s**: Parameters controlling the mobility of the phases $\phi$ and $\psi$.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+  .. math::
+
+     \gamma_{mn} = \frac{\sqrt{(\kappa_m + \kappa_n)(\lambda_m + \lambda_n)}}{6}
```
```suggestion
     \gamma_{mn} = \frac{\alpha}{6}\left(\kappa_m + \kappa_n\right)
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+     \gamma_{mn} = \frac{\sqrt{(\kappa_m + \kappa_n)(\lambda_m + \lambda_n)}}{6}
+
+* **alpha**: Gradient energy coefficient.
```
```suggestion
* **alpha**: Parameter correlated with the interface width in LB units. The interface width is given by 4*alpha.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+The *dumpxdmf* keyword enables output of the fluid fields to files that can be visualized using Paraview or other XDMF-compatible visualization software. The output includes:
+
+* Density field ρ (converted to physical units)
+* Order parameters φ and ψ (converted to physical units)
```
```suggestion
* Order parameters \phi and \psi
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The *dumpxdmf* keyword enables output of the fluid fields to files that can be visualized using Paraview or other XDMF-compatible visualization software. The output includes:
+
+* Density field ρ (converted to physical units)
```
```suggestion
* Density field ρ
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* Density field ρ (converted to physical units)
+* Order parameters φ and ψ (converted to physical units)
+* Pressure field p (converted to physical units)
```
```suggestion
* Pressure field p
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+* Density field ρ (converted to physical units)
+* Order parameters φ and ψ (converted to physical units)
+* Pressure field p (converted to physical units)
+* Velocity field **u** (converted to physical units)
```
```suggestion
* Velocity vector field **u**
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+**Output**
+
+The *dumpxdmf* keyword enables output of the fluid fields to files that can be visualized using Paraview or other XDMF-compatible visualization software. The output includes:
```
```suggestion
The *dumpxdmf* keyword enables output of the fluid fields to files that can be visualized using Paraview or other XDMF-compatible visualization software. The output (in LB units) includes:
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+* A text .xdmf file containing metadata and grid structure
+* A binary .raw file containing the actual field data
+
+The concentration fields can be recovered from the output using:
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+.. math::
+
+   C_1 = \frac{\rho + \phi}{2}, \quad C_2 = \frac{\rho - \phi}{2}, \quad C_3 = \psi
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The concentration fields can be recovered from the output using:
+
+.. math::
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+**Boundary Conditions**
+
+The ternary lattice-Boltzmann implementation currently supports periodic boundary conditions in all three directions. The boundary conditions are specified through the main LAMMPS :doc:`boundary <boundary>` command and must be set to periodic (p p p) for proper operation.
```
```suggestion
The ternary lattice-Boltzmann implementation currently only supports periodic boundary conditions in all three directions. The boundary conditions are specified through the main LAMMPS :doc:`boundary <boundary>` command and must be set to periodic (p p p) for proper operation.
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The ternary lattice-Boltzmann implementation currently supports periodic boundary conditions in all three directions. The boundary conditions are specified through the main LAMMPS :doc:`boundary <boundary>` command and must be set to periodic (p p p) for proper operation.
+
+*Periodic Boundaries*
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+*Periodic Boundaries*
+
+With periodic boundary conditions, the fluid domain wraps around in all directions. The distribution functions that stream out of one side of the simulation box re-enter from the opposite side. This is handled automatically through the MPI communication scheme, which includes a halo region of 2 lattice layers on each side of the local processor subdomain.
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+With periodic boundary conditions, the fluid domain wraps around in all directions. The distribution functions that stream out of one side of the simulation box re-enter from the opposite side. This is handled automatically through the MPI communication scheme, which includes a halo region of 2 lattice layers on each side of the local processor subdomain.
+
+The halo communication ensures that:
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+*Periodic Boundaries*
+
+With periodic boundary conditions, the fluid domain wraps around in all directions. The distribution functions that stream out of one side of the simulation box re-enter from the opposite side. This is handled automatically through the MPI communication scheme, which includes a halo region of 2 lattice layers on each side of the local processor subdomain.
+
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+The halo communication ensures that:
+
+* Gradient calculations (∇ρ, ∇φ, ∇ψ) required for the chemical potentials can be performed correctly near subdomain boundaries
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+The halo communication ensures that:
+
+* Gradient calculations (∇ρ, ∇φ, ∇ψ) required for the chemical potentials can be performed correctly near subdomain boundaries
+* Laplacian calculations (∇²ρ, ∇²φ, ∇²ψ) have access to necessary neighbor information
```
```suggestion
```

akgokulraman commented on doc/src/fix_lb_multicomponent.rst:
```diff
@@ -0,0 +1,245 @@
+
+* Gradient calculations (∇ρ, ∇φ, ∇ψ) required for the chemical potentials can be performed correctly near subdomain boundaries
+* Laplacian calculations (∇²ρ, ∇²φ, ∇²ψ) have access to necessary neighbor information
+* Streaming of distribution functions across processor boundaries is handled correctly
```
```suggestion
```

akgokulraman reviewed (changes requested):
Please make the following changes

rzaman64 committed:
Update doc/src/fix_lb_multicomponent.rst

Co-authored-by: Gokul Raman <53153074+akgokulraman@users.noreply.github.com>

rzaman64 committed:
Update doc/src/fix_lb_multicomponent.rst

Co-authored-by: Gokul Raman <53153074+akgokulraman@users.noreply.github.com>
